### PR TITLE
83 establish basic pattern for tests

### PR DESCRIPTION
**Pull request by murny**
Whats currently in here:

- Add config for rspec
- Fix generated tests from curation concerns
- FactoryGirl/Capaybara/etc
- Add travis
- Add coveralls

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,63 +1,79 @@
...
-    - 'lib/sufia/arkivo/metadata_munger.rb'
+  Max: 45 # default 15
 
+Metrics/ClassLength:
+  Max: 500 # default 100
+
+# A complexity metric that is strongly correlated to the number
+# of test cases needed to validate a method.
+Metrics/CyclomaticComplexity:
+  Max: 17 # default 6
```
Comment (mbarnett) [inline .rubocop.yml]: Are we just making up a number here? Is this worth keeping? I'm not a huge fan of cyclometric complexity tests tbh. It tends to whine about a lot of idiomatic constructs.

File: .rubocop.yml [hunk 2]
```diff
@@ -1,63 +1,79 @@
...
+
+Metrics/LineLength:
+  Enabled: true
+  Max: 120 # default is 80
+
+# Avoid methods longer than 10 lines of code.
 Metrics/MethodLength:
-  Max: 15
+  Max: 40 # default 10
```
Comment (mbarnett) [inline .rubocop.yml]: 40 seems excessive? 20 Maybe? What's the longest method we have that isn't a mess?

File: .rubocop.yml [hunk 3]
```diff
@@ -1,63 +1,79 @@
...
-  Exclude:
-    - 'app/models/sufia/uploaded_file.rb'
+Metrics/ModuleLength:
+  Max: 500 # default 100
 
-Style/IndentationConsistency:
-  EnforcedStyle: rails
+# A complexity metric geared towards measuring complexity for a human reader.
+Metrics/PerceivedComplexity:
+  Max: 18 # default 7
```
Comment (mbarnett) [inline .rubocop.yml]: Same comment as the cyclometric complexity I guess. I'd rather have someone say "hey, this is too complex/dense for me to read" than have a bunch of heuristics half-ass the sentiment.

**Review by mbarnett — COMMENTED** [review]
Looks fine other than the notes on a few of the cops.

File: .rubocop.yml (hunk 1 above)
Comment (murny) [reply .rubocop.yml]: Yeah just buffed these cops up. We can turn them off if you want.

File: .rubocop.yml (hunk 3 above)
Comment (murny) [reply .rubocop.yml]: Can get behind that!

File: .rubocop.yml (hunk 2 above)
Comment (murny) [reply .rubocop.yml]: In current hydranorth we have some of our rake tasks over 500+ (migration_audit has a 641 method). With rake tasks aside it seems like 42 (batch update job) or 48 (bin/bin/fix.rb)?

File: Gemfile [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
+# hidden Sufia dependencies below
 # Sufia pushes flipflop on the app as its not an actual gem, should PR this back somehow?
 gem 'flipflop', git: 'https://github.com/jcoyne/flipflop.git', branch: 'hydra'
 gem 'jbuilder', '~> 2.0' # Hidden Sufia Dependencies...Need PR against sufia...
 # gem 'rsolr', '~> 1.0' # Do we need this? Comment it out for now
 
-# fail update on hydra...was removes in curation concerns...but sufia still needs this
-gem 'blacklight_advanced_search', '~> 6.0'
+# fail on sufia 7.2...they have a lose dependency on CC...which is breaking if it jumps to a higher version
+gem 'curation_concerns', '1.6.1'
```
Comment (murny) [inline Gemfile:17]: Need to lock this down in our app, as Sufia 7.2 is currently not doing this. Causing a bunch of issues if CC jumps to 1.7.x versions.

**Review by mbarnett — APPROVED** [review]
👍

**Pull request merged**